### PR TITLE
Update Rubocop to 0.34.2

### LINKED ISSUE #903: bump Rubocop version to 0.33
Any chance we could bump the Rubocop version to the latest? i.e. 0.33.  From the Gemfile it looks like hound is on "0.29.1"

There are various config options in 0.33 such as DisabledByDefault, which I think might help us get round the various confusions arising from the way in which hound overrides the rubocop settings in each project:

https://github.com/thoughtbot/hound/issues/576

Many thanks in advance

### PR DESCRIPTION
**Pull request by thorncp**
* Renames `Style/Blocks` to `Style/BlockDelimiters`
* Renames `Style/UnneededPercentX` to `Style/CommandLiteral`
* Disables all new cops for Hound and thoughtbot config.
* Does not use the new `DisabledByDefault` config value, which only
  enables cops you specify. I left this off because we have all the cops
  explicitly defined in the config already.
* Updates expected violation messages to reflect new text
* Tweaks a couple tests that were no longer seeing the
  "Unnecessary spacing detected." message. I'm not sure what this was
  even covering, given that other cops in those tests are completely
  handling the whitespace (I think).
* `Style/RegexpLiteral` no longer supports a `MaxSlashes` parameter, so
  it's removed.

Will close #903

Potentially solves #725

### CODE REVIEW

File: config/style_guides/ruby.yml [hunk 1]
```diff
@@ -20,7 +20,12 @@ AllCops:
   - "db/schema.rb"
   RunRailsCops: false
   DisplayCopNames: false
+  DisplayStyleGuide: false
   StyleGuideCopsOnly: false
+  DisabledByDefault: false
```
Comment (gylaz) [inline config/style_guides/ruby.yml]: What is this set to by default in RuboCop?

File: config/style_guides/ruby.yml [hunk 2]
```diff
@@ -923,7 +1010,11 @@ Style/UnneededPercentQ:
   Description: Checks for %q/%Q when single quotes or double quotes would do.
   StyleGuide: https://github.com/bbatsov/ruby-style-guide#percent-q
   Enabled: true
-Style/UnneededPercentX:
```
Comment (gylaz) [inline config/style_guides/ruby.yml]: This was removed (or rather became `CommandLiteral`) but what about its options?

**Comment by gylaz** [comment]
Thoughts on **not** disabling new rules? They are fairly uncommon violation rules, and it might give us a reality check about the config changes we are doing -- given that in the future we'll just be rolling with linter defaults, thus each time we'll update the rules will change on our users.

File: config/style_guides/ruby.yml (hunk 1 above)
Comment (thorncp) [inline config/style_guides/ruby.yml]: [false](https://github.com/bbatsov/rubocop/blob/master/config/default.yml#L51)

File: config/style_guides/ruby.yml (hunk 2 above)
Comment (thorncp) [inline config/style_guides/ruby.yml]: The old options are in the same place, but looks like I added two `CommandLiteral`s by accident. Removed one in b36763283a0c0a8ca7c228839e12b67360328f69

**Comment by thorncp** [comment]
I'm fine using the Rubocop defaults for all the new cops. I just went into this PR with thought of having the least resistance.

**Comment by thorncp** [comment]
Pulled in Rubocop defaults for new inters in a6c8bda29d5c7e5ce432231c0c738f8b5e706d27

**Comment by gylaz** [comment]
Looks good to me. @salbertson Thoughts?

**Comment by teoljungberg** [comment]
Looks good to me

**Comment by salbertson** [comment]
Looks good to me! :+1:

**Pull request merged**